### PR TITLE
[WIP] added feature heatmaps of one-hot encoded and similarity encoded

### PR DESCRIPTION
**Pull request by pierreglaser**

### CODE REVIEW

File: examples/plot_investigating_dirty_categories.py [hunk 1]
```diff
@@ -122,24 +123,50 @@
 f2.tight_layout()
 
 ########################################################################
-# As shown in the previous plot, we see that "communication Equipment technician"'s
-# nearest neighbor is "telecommunication technician", although it is also
+# As shown in the previous plot, we see that the nearset neighbor of "Communication Equipment Technician"
```
Comment (GaelVaroquaux) [inline examples/plot_investigating_dirty_categories.py]: Typo at "nearest". Also, this line is too long.

File: examples/plot_investigating_dirty_categories.py [hunk 2]
```diff
@@ -122,24 +123,50 @@
...
 
+f4, ax4 = plt.subplots(figsize=(6, 6))
+cax4 = ax4.matshow(transformed_values[:n_obs, :n_obs])
+f4.colorbar(cax4)
+f4.suptitle('Employee Position Title values, similarity encoded')
+ax4.xaxis.tick_bottom()
+f4.tight_layout()
+
+#########################################################################
+# you can check out how similarity encoding impacts prediction performance
```
Comment (GaelVaroquaux) [inline examples/plot_investigating_dirty_categories.py]: I would rephrase as following:

"Other examples in the dirty_cat documentation show how similarity encoding impacts prediction performance."

File: examples/plot_investigating_dirty_categories.py [hunk 3]
```diff
@@ -122,24 +123,51 @@
...
 #########################################################################
 # Encoding categorical data using SimilarityEncoder
 # -------------------------------------------------
 #
 # A typical data-science workflow uses one-hot encoding to represent
 # categories.
+from sklearn.preprocessing import CategoricalEncoder
+
+n_obs, n_categories = 20, 20
+# employee_position_titles = values[
```
Comment (pierreglaser) [inline examples/plot_investigating_dirty_categories.py]: Right now, we use unique sorted values as entries to the one hot encoder, to the transformed values are the identity matrix.
the 4 commented lines pick observations from non-unique non sorted values at random, so we will get a more realistic one-hot encoded matrix (the columns won't necessarily sum to 1, and the 'hot' values will be scattered everywhere in the matrix, not only in the diagonal)
Thoughts @GaelVaroquaux?

Comment (GaelVaroquaux) [reply examples/plot_investigating_dirty_categories.py]: I don't think that we should use the sorted values: it does not reflect the usage. We should fit-transform a "head" of the data matrix.

File: examples/plot_investigating_dirty_categories.py [hunk 4]
```diff
@@ -122,24 +123,51 @@
...
+#     'Employee Position Title'].sort_values().to_frame()
+# or
+# employee_position_titles=values[
+#     'Employee Position Title'].sample(n=n_observations)?
```
Comment (GaelVaroquaux) [inline examples/plot_investigating_dirty_categories.py]: Why not "head"?

Comment (pierreglaser) [reply examples/plot_investigating_dirty_categories.py]: head works fine too.
